bciers/apps/compliance/src/app/components/compliance-summaries/fetchComplianceSummariesPageData.ts

```
@@ -0,0 +1,23 @@
+import { ComplianceSummary } from "./types";
+import { actionHandler } from "@bciers/actions";
+import buildQueryParams from "@bciers/utils/src/buildQueryParams";
+
+export const fetchComplianceSummariesPageData = async (params: {
+  [key: string]: any;
+}): Promise<{
+  items: ComplianceSummary[];
+  count: number;
+}> => {
+  const queryParams = buildQueryParams(params);
+
+  const data = await actionHandler(
+    `compliance/summaries${queryParams}`,
+    "GET",
+    "",
+  );
+
+  return {
+    items: data?.items || [],
+    count: data?.count || 0,
+  };
```

I feel this might swallow an error from the server if the data returned is malformatted or wrong, I would rather the frontend to crash or throw an informative error if the `data` object has issues

---

Sounds good, I will make changes to align with how other components in the codebase handle errors from the actionHandler, ensuring that malformed or error responses will properly surface to the UI rather than being silently handled.